public/app/features/dashboard-scene/scene/DashboardScene.tsx

```diff
@@ -359,32 +363,50 @@ export class DashboardScene extends SceneObjectBase<DashboardSceneState> {
       (event: SceneObjectStateChangedEvent) => {
         if (event.payload.changedObject instanceof SceneRefreshPicker) {
           if (Object.prototype.hasOwnProperty.call(event.payload.partialUpdate, 'intervals')) {
-            this.setIsDirty();
+            this.detectChanges();
           }
         }
         if (event.payload.changedObject instanceof SceneGridItem) {
-          this.setIsDirty();
+          this.detectChanges();
         }
         if (event.payload.changedObject instanceof DashboardScene) {
           if (Object.keys(event.payload.partialUpdate).some((key) => PERSISTED_PROPS.includes(key))) {
-            this.setIsDirty();
+            this.detectChanges();
           }
         }
         if (event.payload.changedObject instanceof SceneTimeRange) {
-          this.setIsDirty();
+          this.detectChanges();
         }
         if (event.payload.changedObject instanceof DashboardControls) {
           if (Object.prototype.hasOwnProperty.call(event.payload.partialUpdate, 'hideTimeControls')) {
-            this.setIsDirty();
+            this.detectChanges();
           }
         }
+        if (event.payload.changedObject instanceof SceneVariableSet) {
+          this.detectChanges();
+        }
+        if (event.payload.changedObject instanceof DashboardAnnotationsDataLayer) {
+          this.detectChanges();
+        }
+        if (isSceneVariableInstance(event.payload.changedObject)) {
+          this.detectChanges();
+        }
       }
     );
   }
 
-  private setIsDirty() {
-    if (!this.state.isDirty) {
-      this.setState({ isDirty: true });
+  private detectChanges() {
+    const { hasChanges, hasTimeChanges, hasVariableValueChanges } = getSaveDashboardChange(this, true, true);
+    const hasChangesToSave = hasChanges || hasTimeChanges || hasVariableValueChanges;
```

Comment:
if you pass true for saveTimeRange and saveVariableValues there is no need to check hasTimeChanged, hasVariableValue changed etc here. hasChanges will be all you need to check, so hasChangesToSave is not needed here, and you can ignore hasTimeChanged and hasVariableValueChanged. 

but for drity state I think we can actually always ignore time & variable value changes

Reply:
oh, is this on purpose? I thought when the button was dimmed, the idea was to not allow to save them and now was possible because there was a bug there. Alexa was thinking the same that I was 🤔